### ISSUE TITLE
Reordering of multiple filesets is not reflected in viewer

### ISSUE THREAD

**Issue by carakey** [OregonDigital/OD2#2863]
### Descriptive summary

_Background: The Metadata Team discussed and agreed that, in certain cases, it will be preferable to create single works with multiple filesets rather than true complex objects._

Multiple filesets attached to a work can be managed using the File Manager option in the UI: setting which is the Thumbnail and the Representative Media for the work, and applying alphabetical sorting. However, changes to order made through the File Manager are not mirrored in the UV for the work.  

In this example on Staging -- https://staging.oregondigital.org/concern/images/q524jn80f?locale=en -- the black and white image was added first and the color image added second. After initially saving, I used the File Manager to edit the files: I set the color image as the thumbnail, and renamed the files from "_front" and "_back" to "a" and "b" so that I could sort them alphabetically. After that edit, the order is shown correctly at the bottom of the work, and the desired thumbnail appears in search results, etc, but the black and white image remains first in the viewer. I cannot modify this order elsewhere in the UI.

![image](https://github.com/OregonDigital/OD2/assets/17655103/7e933fe5-2931-4282-9f35-ad3a13303c88)

### Expected behavior

Multiple files attached to a single work can be reordered using the File Manager, and those changes will apply to the order that files are displayed in the viewer.

Having additional sorting controls in the File Manager, beyond initial upload order and alphabetical, would be an enhancement.

### Related work

Link to related tickets or prior related work here.

### Accessibility Concerns

Add any information here to indicate any known or suspected accessibility issues for this ticket

**Comment by CGillen** [comment]
This is likely related to #2786. I'm keeping them separate issues, but if you're working on this take a look at that one and see if it's a trivial fix

**Comment by KevinJonesMeta** [comment]
Displayed titles in contents list when I opened example work above were the original file names. After going into the work form and saving it without making changes the displayed titles for objects in the contents list changed from the filename to the name of the work title. Order of objects in contents list did not change (still out of desired order).

**Comment by KevinJonesMeta** [comment]
QA fail. After changing filename on color image to end with "c" instead of "a" and leaving black and white image file ending in "b" the color image displayed in the viewer contents list first but second in the thumbnail list at the bottom of the work show page. Attempts to duplicate that order after returning to "a"-"b" order have failed. Moved back to Ready.

**Comment by KevinJonesMeta** [comment]
Also, I checked the .nt file before and after changing the file names and these n-triples were the same in both. I don't know if it is relevant but it looks like it should have changed when I changed the order of files based on the predicates?
<https://staging.oregondigital.org/concern/images/q524jn80f/list_source> <http://purl.org/dc/terms/hasPart> <https://staging.oregondigital.org/concern/images/q524jn80f/list_source#g19464120> .
<https://staging.oregondigital.org/concern/images/q524jn80f/list_source> <http://purl.org/dc/terms/hasPart> <https://staging.oregondigital.org/concern/images/q524jn80f/list_source#g19467160> .
<https://staging.oregondigital.org/concern/images/q524jn80f/list_source> <http://www.iana.org/assignments/relation/last> <https://staging.oregondigital.org/concern/images/q524jn80f/list_source#g19467160> .
<https://staging.oregondigital.org/concern/images/q524jn80f/list_source> <http://www.iana.org/assignments/relation/first> <https://staging.oregondigital.org/concern/images/q524jn80f/list_source#g19464120> .

**Comment by carakey** [comment]
Still QA fail. I'm not seeing any change in behavior. 

QA'd using the [work](https://staging.oregondigital.org/concern/images/q524jn80f?locale=en) linked in the original comment. In the File Manager, I renamed the files and re-saved. The color image is listed first and is selected as featured/thumbnail. The viewer still shows it after the B&W image file.

**Comment by carakey** [comment]
QA with a new work:

- I created a new work [here](https://staging.oregondigital.org/concern/images/5h73pw14h?locale=en). I uploaded two images to the work: `2022002002b.tif` (black & white, should appear last, refer to as "File B"), followed by `2022002002a.tif` (color, should appear first, refer to as "File A"). I saved, approved, and waited for derivatives to generate. With the test work live/public, File B appears first in the viewer and File A appears second -- as expected given the upload order.
- In the File Manager I clicked "Sort Alphabetically" and set File A to be the Thumbnail and Representative Media for the work, then clicked Save.
- I returned to the work show page and the images appear in the correct, updated order with File A followed by File B. 

__QA pass!__
